Design.md

```diff
@@ -192,33 +192,23 @@ __. and ..__
 File creation is decoupled from open and write. A new entry is created in DBD. the metadata associated to the object notably contains:
  - a random content-md5 with char #32 set to '-'
  - Date and last-modified set to now()
- - owner-display-name and owner-id set to configured values
+ - owner-display-name and owner-id set to the values found in the bucket attributes
  - an empty locations array
- - Future: Do we have to remove the placeholder??
+ - Future: Do we have to remove or update the placeholder?
 
 This has the effect of creating instantly an empty file.
 
-
-## File removal
-
- - Future: If the object is the last with this prefix, create a placeholder
- - delete object entry from DBD bucket
- - delete all parts from sproxyd storage
-
-At this time, it's kind of broken when the file to remove is the last in the prefix.
-
 ### Directory creation
 
- - Future: Create a placeholder for the directory
-
-At this time, directory creation is badly implemented using internal states that can't be relied on.
+Directories are created as placeholders. A placeholder is an empty object named after the directory name with a trailing slash (```/```) with content-type set to ```application/x-directory```
 
 
-### Directory removal
+## File/Directory removal
 
- - Future: delete the placeholder if it is the last object with this prefix
-
-Today, it's almost broken
+ - create the placeholder and replace it if it already exists ! it is faster than a lookup
```

Comment:
On the removal of the last object of a given prefix, it would be unexpected from the NFS point of view to loose the corresponding directory for this prefix.